storage/rangefeed: TestProcessorSlowConsumer failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/395d842feb97c5bd8cad2b32b71a5156c03061eb

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestProcessorSlowConsumer PKG=github.com/cockroachdb/cockroach/pkg/storage/rangefeed TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1116894&tab=buildLog

```
=== RUN   TestProcessorSlowConsumer
==================
WARNING: DATA RACE
Write at 0x00c000280328 by goroutine 29:
  github.com/cockroachdb/cockroach/pkg/util/interval.(*llrbTree).Delete()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/interval/llrb_based_interval.go:371 +0x3a8
  github.com/cockroachdb/cockroach/pkg/storage/rangefeed.(*registry).Unregister()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/rangefeed/registry.go:376 +0x6f
  github.com/cockroachdb/cockroach/pkg/storage/rangefeed.(*Processor).Start.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/rangefeed/processor.go:234 +0xf3e
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:198 +0x145

Previous read at 0x00c000280328 by goroutine 28:
  github.com/cockroachdb/cockroach/pkg/util/interval.(*llrbTree).Len()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/interval/llrb_based_interval.go:190 +0x3e
  github.com/cockroachdb/cockroach/pkg/storage/rangefeed.(*registry).Len()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/rangefeed/registry.go:325 +0x50
  github.com/cockroachdb/cockroach/pkg/storage/rangefeed.TestProcessorSlowConsumer.func4()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/rangefeed/processor_test.go:467 +0x6c
  github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:41 +0x41
  github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:188 +0xf7
  github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon()
      /go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:47 +0x138
  github.com/cockroachdb/cockroach/pkg/storage/rangefeed.TestProcessorSlowConsumer()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/rangefeed/processor_test.go:465 +0xd45
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:827 +0x162

Goroutine 29 (running) created at:
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:191 +0xc3
  github.com/cockroachdb/cockroach/pkg/storage/rangefeed.(*Processor).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/rangefeed/processor.go:169 +0x151
  github.com/cockroachdb/cockroach/pkg/storage/rangefeed.newTestProcessorWithTxnPusher()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/rangefeed/processor_test.go:141 +0x376
  github.com/cockroachdb/cockroach/pkg/storage/rangefeed.newTestProcessor()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/rangefeed/processor_test.go:146 +0x4a
  github.com/cockroachdb/cockroach/pkg/storage/rangefeed.TestProcessorSlowConsumer()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/rangefeed/processor_test.go:379 +0x89
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:827 +0x162

Goroutine 28 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:878 +0x659
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1119 +0xa8
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:827 +0x162
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1117 +0x4ee
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1034 +0x2ee
  main.main()
      _testmain.go:78 +0x221
==================
I190130 08:41:35.290748 26 util/stop/stopper.go:536  quiescing; tasks left:
1      [async] rangefeed: output loop
--- FAIL: TestProcessorSlowConsumer (0.02s)
    testing.go:771: race detected during execution of test

```

--- a/pkg/storage/rangefeed/processor_test.go
+++ b/pkg/storage/rangefeed/processor_test.go
@@ -463,8 +463,8 @@ func TestProcessorSlowConsumer(t *testing.T) {
 	require.Equal(t, testProcessorEventCCap+1, len(r1Stream.Events()))
 	require.Equal(t, newErrBufferCapacityExceeded().GoError(), (<-r1ErrC).GoError())
 	testutils.SucceedsSoon(t, func() error {
-		if p.Len() != 1 {
-			return fmt.Errorf("processor had %d regs, wanted %d", p.reg.Len(), 1)
+		if act, exp := p.Len(), 1; exp != act {
+			return fmt.Errorf("processor had %d regs, wanted %d", act, exp)
 		}
 		return nil
 	})

